alacritty_terminal/src/term/cell.rs

```diff
@@ -128,6 +182,28 @@ impl Cell {
     pub fn underline_color(&self) -> Option<Color> {
         self.extra.as_ref()?.underline_color
     }
+
+    /// Set hyperlink.
+    pub fn set_hyperlink(&mut self, hyperlink: Option<Hyperlink>) {
+        let drop_hyperlink = hyperlink.is_none()
+            && self.extra.as_ref().map_or(true, |extra| {
+                !extra.zerowidth.is_empty() || extra.underline_color.is_some()
+            });
+
+        if drop_hyperlink {
+            self.extra = None;
+            return;
+        }
+
+        let extra = self.extra.get_or_insert(Default::default());
+        Arc::make_mut(extra).hyperlink = hyperlink;
```

Comment:
```suggestion
        if drop_hyperlink {
            self.extra = None;
        } else {
            let extra = self.extra.get_or_insert(Default::default());
            Arc::make_mut(extra).hyperlink = hyperlink;
        }
```

Should avoid the extra return, just complicates control flow.